notebooks/vertex_genai/solutions/langchain_components_part_1.ipynb

```diff
@@ -0,0 +1,1293 @@
+{
```

Comment:
<p>A brief explanation like the following could be useful here: "We will define a rate limiter to control the number of queries per minute issued to the API. Then we will derive a custom class from the parent class VertexAIEmbeddings in order to override its embed_documents method to wrap the API calls within that rate limiter."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/482/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>+1</p><p> </p><p> </p><p> </p>

Reply:
<p>Done, thank you! </p>